### PR TITLE
Allowing for search algorithm overriding

### PR DESCRIPTION
**Pull request by moniika**
- renaming `search` to `searchAndHighlight` AND `populateBlocks` to `search`
- adding function `setSearchFunction` to allow for overriding default search behavior (and logic in to use the custom search if available).
- changing `Block.Block` to `Block.BlockSvg`
- moving dom creation to `createDom` method

### CODE REVIEW

File: blockly-ws-search/src/WorkspaceSearch.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ export class WorkspaceSearch {
 
     /**
      * A list of blocks that came up in the search
-     * @type {!Array.<Blockly.Block>}
+     * @type {!Array.<Blockly.BlockSvg>}
```
Comment (alschmiedt) [inline blockly-ws-search/src/WorkspaceSearch.js:45]: Why the change to BlockSvg?

File: blockly-ws-search/src/WorkspaceSearch.js [hunk 2]
```diff
@@ -68,6 +68,13 @@ export class WorkspaceSearch {
      */
     this.searchText_ = '';
 
+    /**
+     * An optional custom search function to override default behavior.
+     * @type {?function(Blockly.WorkspaceSvg, string): Array.<Blockly.BlockSvg>}
+     * @protected
+     */
+    this.customSearchFunction_ = null;
```
Comment (alschmiedt) [inline blockly-ws-search/src/WorkspaceSearch.js:76]: We might want to make this static. Blockly.WorkspaceSearch.customSearchFunction. Doesn't seem like there would be a large use case for changing on object by object basis.

File: blockly-ws-search/src/WorkspaceSearch.js [hunk 3]
```diff
@@ -409,27 +422,40 @@ export class WorkspaceSearch {
...
+  }
+
+  /**
+   * Searches the workspace for the current search term and highlights found
+   * blocks.
+   * @param {Blockly.BlockSvg=} opt_block Block to use as the current
    *     block if it is included in new search blocks.
    */
-  search(opt_block) {
+  searchAndHighlight(opt_block) {
```
Comment (alschmiedt) [inline blockly-ws-search/src/WorkspaceSearch.js:440]: Could this be private? Since we now allow them to set the search function, and they can change how the block is highlighted through css?

Comment (moniika) [reply blockly-ws-search/src/WorkspaceSearch.js:440]: Leaving this public gives the freedom of triggering/binding it outside of the search bar.

File: blockly-ws-search/src/WorkspaceSearch.js (hunk 1 above)
Comment (moniika) [reply blockly-ws-search/src/WorkspaceSearch.js:45]: Keeping track Block vs BlockSvg started to seem like too much overhead, especially after introducing a way for users to custom set the search function (and we are expecting the blocks returned to be BlockSvg). Since we take in a WorkspaceSvg, we're only sort of supporting non-rendered workspaces. We should probably talk more offline.
